packages/central-server/src/apiV2/import/importSurveyResponses/importSurveyResponses.js

```diff
@@ -168,32 +168,56 @@ export async function importSurveyResponses(req, res) {
       for (let columnIndex = minSurveyResponseIndex; columnIndex <= maxColumnIndex; columnIndex++) {
         const columnHeader = getColumnHeader(sheet, columnIndex);
         const importMode = getImportMode(columnHeader);
+        const entityCode = getInfoForColumn(sheet, columnIndex, 'Entity Code');
+        const entityName = getInfoForColumn(sheet, columnIndex, 'Entity Name');
+        const entity = await models.entity.findOne({ code: entityCode });
+
+        if (entityCode && entityName) {
+          if (entity) {
+            if (entity?.name !== entityName) {
+              throw new Error(
+                `Entity code and name don't match: ${entity?.name} and ${entityName}`,
+              );
+            }
+          } else {
+            throw new Error(`Entity code does match any existing entity: ${entityCode}`);
```

Comment:
```suggestion
        if (entityCode && entityName) {
           if(!entity) {
                 throw new Error(`Entity code does match any existing entity: ${entityCode}`)
           }
           
           if (entity.name !== entityName) {
              throw new Error(
                `Entity code and name don't match: ${entity?.name} and ${entityName}`,
              );
            }
        
        }
```

Just a slight rearranging of this one - it's nice to check if the thing (in this case `entity`) exists before doing anything else. It reduces the need for an `else` statement, and makes it generally more readable. It also means you always know `entity.name` will exist, so you don't need to handle it with a `?`